eclair-core/src/test/scala/fr/acinq/eclair/payment/receive/InvoicePurgerSpec.scala

```diff
@@ -0,0 +1,91 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.payment.receive
+
+import akka.actor.typed.scaladsl.adapter.ClassicActorSystemOps
+import akka.testkit.TestProbe
+import fr.acinq.bitcoin.Block
+import fr.acinq.eclair.TestDatabases.TestPgDatabases
+import fr.acinq.eclair.db.{IncomingPayment, IncomingPaymentStatus, PaymentType, PaymentsDbSpec}
+import fr.acinq.eclair.payment.Bolt11Invoice
+import fr.acinq.eclair.payment.receive.InvoicePurger.{PurgeCompleted, PurgeEvent}
+import fr.acinq.eclair.{CltvExpiryDelta, MilliSatoshiLong, TestKitBaseClass, TimestampMilli, TimestampMilliLong, TimestampSecond, TimestampSecondLong, randomBytes32}
+import org.scalatest.funsuite.AnyFunSuiteLike
+
+import scala.concurrent.duration.DurationInt
+
+class InvoicePurgerSpec extends TestKitBaseClass with AnyFunSuiteLike {
+
+  import PaymentsDbSpec._
+
+  test("purge invoices") {
+    val dbs = TestPgDatabases()
+    val db = dbs.db.payments
+    val count = 10
+
+    // create expired invoices
+    val expiredInvoices = Seq.fill(count)(Bolt11Invoice(Block.TestnetGenesisBlock.hash, Some(100 msat), randomBytes32(), alicePriv, Left("expired invoice"), CltvExpiryDelta(18),
+      timestamp = 1 unixsec))
+    val expiredPayments = expiredInvoices.map( invoice => IncomingPayment(invoice, randomBytes32(), PaymentType.Standard, invoice.createdAt.toTimestampMilli, IncomingPaymentStatus.Expired))
+    expiredInvoices.lazyZip(expiredPayments).foreach((invoice, payment) => db.addIncomingPayment(invoice, payment.paymentPreimage))
+
+    // create pending invoices
+    val pendingInvoices = Seq.fill(count)(Bolt11Invoice(Block.TestnetGenesisBlock.hash, Some(100 msat), randomBytes32(), alicePriv, Left("pending invoice"), CltvExpiryDelta(18),
+      timestamp = TimestampSecond.now() - 600))
+    val pendingPayments = pendingInvoices.map( invoice => IncomingPayment(invoice, randomBytes32(), PaymentType.Standard, invoice.createdAt.toTimestampMilli, IncomingPaymentStatus.Pending))
+    pendingInvoices.lazyZip(pendingPayments).foreach((invoice, payment) => db.addIncomingPayment(invoice, payment.paymentPreimage))
+
+    // create paid invoices
+    val receivedAt = TimestampMilli.now() + 1.milli
+    val paidInvoices = Seq.fill(count)(Bolt11Invoice(Block.TestnetGenesisBlock.hash, Some(100 msat), randomBytes32(), alicePriv, Left("paid invoice"), CltvExpiryDelta(18),
+      timestamp = TimestampSecond.now()))
+    val paidPayments = paidInvoices.map( invoice => IncomingPayment(invoice, randomBytes32(), PaymentType.Standard, invoice.createdAt.toTimestampMilli, IncomingPaymentStatus.Received(100 msat, receivedAt)))
+    paidInvoices.lazyZip(paidPayments).foreach((invoice, payment) => { db.addIncomingPayment(invoice, payment.paymentPreimage)
```

Comment:
nit: it's a bit weird to have `db.addIncomingPayment` on the same line since there are other statements in that block, can you put it on the line below?

Reply:
I ran ctrl + alt + L on this one too, which fixed that and some spacing issues.